modules/google-analytics/classes/wp-google-analytics-universal.php

```diff
@@ -65,23 +65,39 @@ public function wp_head() {
 		 * @param array $custom_vars Array of universal Google Analytics queue elements
 		 */
 		$universal_commands = apply_filters( 'jetpack_wga_universal_commands', array() );
-
-		$async_code = "
-			<!-- Jetpack Google Analytics -->
-			<script>
-				window.ga = window.ga || function(){ ( ga.q = ga.q || [] ).push( arguments ) }; ga.l=+new Date;
-				ga( 'create', '%tracking_id%', 'auto' );
-				ga( 'require', 'ec' );
-				%universal_commands%
-			</script>
-			<script async src='https://www.google-analytics.com/analytics.js'></script>
-			<!-- End Jetpack Google Analytics -->
-		";
-		$async_code = str_replace( '%tracking_id%', $tracking_code, $async_code );
-
-		$universal_commands_string = implode( "\r\n", $universal_commands );
-		$async_code = str_replace( '%universal_commands%', $universal_commands_string, $async_code );
-
+		if ( substr( $tracking_code, 0, 2 ) === 'G-' ) {
+			// Upgrade script from UA to GA4 -- https://developers.google.com/analytics/devguides/collection/upgrade/analyticsjs.
+			// phpcs:disable WordPress.WP.EnqueuedResources.NonEnqueuedScript
+			$async_code = "
+				<script async src='https://www.googletagmanager.com/gtag/js?id=%tracking_code%'></script>
+				<script>
+					window.dataLayer = window.dataLayer || [];
+					function gtag(){dataLayer.push(arguments);}
+					gtag('js', new Date());
+					gtag('config', '%tracking_code%');
+
+				</script>
+			";
+			// phpcs:enable
+			$async_code = str_replace( '%tracking_code%', $tracking_code, $async_code );
+		} else {
+			// phpcs:disable WordPress.WP.EnqueuedResources.NonEnqueuedScript
+			$async_code = "
+				<!-- Jetpack Google Analytics -->
```

Comment:
Maybe we could have those comments be there for both implementations?

Reply:
b79ec5525aaa09babc1a4ac48ffc954a4d2e3c33